bindings/wallet-cordova/src/android/WalletPlugin.java

```diff
@@ -179,6 +184,70 @@ private void symmetricCipherDecrypt(final CordovaArgs args, final CallbackContex
         }
     }
 
+    private void settingsNew(final CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
+        final byte[] block0Hash = args.getArrayBuffer(0);
+        final int discriminationInput = args.getInt(1);
+        final JSONObject fees = (JSONObject) args.get(2);
+
+        try {
+            final long constant = Long.parseLong(fees.getString("constant"));
```

Comment:
I don't really know how to do this in a less verbose way, honestly... Besides maybe skipping jni and expose json deserialization from the C api with serde...

Reply:
I'd prefer to keep it verbose rather than making people to read through a bunch of dirty hacks as long as this doesn't affect the performance.